modules/apps/web-experience/portlet-configuration/portlet-configuration-web/src/main/java/com/liferay/portlet/configuration/web/internal/portlet/PortletConfigurationPortlet.java

```diff
@@ -850,7 +850,7 @@ protected PortletPreferences getPortletPreferences(
 
 		PortletPreferencesIds portletPreferencesIds = new PortletPreferencesIds(
 			themeDisplay.getCompanyId(), layout.getGroupId(),
-			PortletKeys.PREFS_OWNER_TYPE_LAYOUT, PortletKeys.PREFS_PLID_SHARED,
+			PortletKeys.PREFS_OWNER_TYPE_GROUP, PortletKeys.PREFS_PLID_SHARED,
```

Comment:
Yes, I think it makes sense. If we find an easy way to change it without requiring an upgrade we will.

Unfortunately this is a part of the code in which things end up being much harder than it seems at first, which is my guess for why it was done in this seemingly weird way.

One of the reasons with our triage, apart from fixing the bugs, is to find potential refactorings to prevent similar bugs from happening in the future.